### PR TITLE
Add Dockerfile and `build-controller-image` Makefile rule

### PR DESCRIPTION
**Pull request by A-Hilaly**
Issue: N/A
Changes:
- Add `Dockerfile`
- Add `build-controller-image` Makefile rule

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jaypipes — COMMENTED** [review]
Hmmm, could we leave this Dockerfile in the main aws/aws-controllers-k8s repository and just update that repo's Makefile and scripts/build-controller-image.sh script to handle the new controller repo locations?

**Comment by A-Hilaly** [comment]
>Hmmm, could we leave this Dockerfile in the main aws/aws-controllers-k8s repository and just update that repo's Makefile and scripts/build-controller-image.sh script to handle the new controller repo locations?

I am just worried about new contributors experience, i think that it will be confusing to have `Dockerfile` outside of the concerned controllers. New contributors will have to clone/use an external repository in order to build the controllers image.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ARG service_alias
+ARG work_dir=/github.com/aws-controllers-k8s/$service_alias-controller
+WORKDIR $work_dir
+# For building Go Module required
+ENV GOPROXY=https://proxy.golang.org,direct
+ENV GO111MODULE=on
+ENV GOARCH=amd64
+ENV GOOS=linux
+ENV CGO_ENABLED=0
+ENV VERSION_PKG=github.com/aws-controllers-k8s/$service_alias/pkg/version
```
[the remark this answers is not in the document]
Comment (A-Hilaly) [reply Dockerfile]: Nice catch! Fixed.

**Comment by A-Hilaly** [comment]
Closing in favor of https://github.com/aws/aws-controllers-k8s/pull/666

**Pull request closed without merging**